### PR TITLE
Attach location to constants

### PR DESCRIPTION
**Pull request by gpetiot**
No diff with test_branch.sh

This allows us to remove all the code that parses the tokens extracted from Source.ml to retrieve the location.
Upstream PR: https://github.com/ocaml/ocaml/pull/10544

### CODE REVIEW

File: vendor/ocaml-4.13/ast_mapper.ml
Comment (gpetiot) [inline vendor/ocaml-4.13/ast_mapper.ml:494]: This should be upstreamed, I will open a PR

**Comment by gpetiot** [comment]
I think we can assume this change won't be upstreamed so it's ready to be reviewed here.

File: vendor/ocaml-4.13-extended/parser.mly
Comment (jberdine) [inline vendor/ocaml-4.13-extended/parser.mly:58]: Nit, but aren't the other functions named `mk<foo>` where the corresponding constructors are `P<foo>`, indicating that `mkconst` would be a more symmetric name than `mkcst`?

**Review by jberdine — APPROVED** [review]
This is a nice simplification! The code looks good to me.

File: vendor/ocaml-4.13-extended/parser.mly
Comment (gpetiot) [reply vendor/ocaml-4.13-extended/parser.mly]: Done!

**Comment by gpetiot** [comment]
I will wait for #1783 to be merged before merging this one

**Review by Julow — APPROVED** [review]
Looks good :)

**Pull request merged**
